From 7c10cc33426138d5547f31b3577d3eeaca1071da Mon Sep 17 00:00:00 2001
From: ching Huang <ching2048@areca.com.tw>
Date: Tue, 6 Oct 2020 15:26:44 +0800
Subject: [PATCH 18/21] scsi: arcmsr: fix warning: right shift count >= width
 of type

Fix warning: right shift count >= width of type.

Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: ching Huang <ching2048@areca.com.tw>
Link: https://lore.kernel.org/r/cbc8c0c56bc82f2abe61f3259ddc6f8c8c9ca53b.camel@areca.com.tw
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 drivers/scsi/arcmsr/arcmsr_hba.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/scsi/arcmsr/arcmsr_hba.c b/drivers/scsi/arcmsr/arcmsr_hba.c
index b0f1272421a7..d4d52d87f89f 100644
--- a/drivers/scsi/arcmsr/arcmsr_hba.c
+++ b/drivers/scsi/arcmsr/arcmsr_hba.c
@@ -655,7 +655,7 @@ static void arcmsr_hbaF_assign_regAddr(struct AdapterControlBlock *acb)
 	/* host buffer low address, bit0:1 all buffer active */
 	writel((uint32_t)(host_buffer_dma | 1), &pmuF->inbound_msgaddr0);
 	/* host buffer high address */
-	writel((uint32_t)(host_buffer_dma >> 32), &pmuF->inbound_msgaddr1);
+	writel(dma_addr_hi32(host_buffer_dma), &pmuF->inbound_msgaddr1);
 	/* set host buffer physical address */
 	writel(ARCMSR_HBFMU_DOORBELL_SYNC1, &pmuF->iobound_doorbell);
 }
@@ -4083,7 +4083,7 @@ static int arcmsr_iop_confirm(struct AdapterControlBlock *acb)
 		acb->msgcode_rwbuffer[4] = acb->ccbsize;
 		dma_coherent_handle = acb->dma_coherent_handle2;
 		cdb_phyaddr = (uint32_t)dma_coherent_handle;
-		cdb_phyaddr_hi32 = (uint32_t)(dma_coherent_handle >> 32);
+		cdb_phyaddr_hi32 = dma_addr_hi32(dma_coherent_handle);
 		acb->msgcode_rwbuffer[5] = cdb_phyaddr;
 		acb->msgcode_rwbuffer[6] = cdb_phyaddr_hi32;
 		acb->msgcode_rwbuffer[7] = acb->completeQ_size;
-- 
2.29.0.rc0

